TestReadFromFileWithoutID is flaky

W0128 18:44:26.920238   14608 config.go:325] Pod[1] (foo.<empty-namespace>) from test failed validation, ignoring: namespace: required value ''
E0128 18:44:26.920811   14608 event.go:160] Could not construct reference to: '&api.BoundPod{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"foo", Namespace:"", SelfLink:"", UID:"foo", ResourceVersion:"", CreationTimestamp:util.Time{Time:time.Time{sec:0, nsec:0x0, loc:(*time.Location)(nil)}}, Labels:map[string]string(nil), Annotations:map[string]string{"kubernetes.io/config.source":""}}, Spec:api.PodSpec{Volumes:[]api.Volume(nil), Containers:[]api.Container(nil), RestartPolicy:api.RestartPolicy{Always:(*api.RestartPolicyAlways)(0xe3d670), OnFailure:(*api.RestartPolicyOnFailure)(nil), Never:(*api.RestartPolicyNever)(nil)}, DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), Host:""}}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'failedValidation' 'Error validating pod foo.<empty-namespace> from test, ignoring: namespace: required value '''
W0128 18:44:27.400243   14608 config.go:325] Pod[1] (foo.<empty-namespace>) from test failed validation, ignoring: namespace: required value ''
E0128 18:44:27.400622   14608 event.go:160] Could not construct reference to: '&api.BoundPod{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"foo", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", CreationTimestamp:util.Time{Time:time.Time{sec:0, nsec:0x0, loc:(*time.Location)(nil)}}, Labels:map[string]string(nil), Annotations:map[string]string(nil)}, Spec:api.PodSpec{Volumes:[]api.Volume(nil), Containers:[]api.Container(nil), RestartPolicy:api.RestartPolicy{Always:(*api.RestartPolicyAlways)(0xe3d670), OnFailure:(*api.RestartPolicyOnFailure)(nil), Never:(*api.RestartPolicyNever)(nil)}, DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), Host:""}}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'failedValidation' 'Error validating pod foo.<empty-namespace> from test, ignoring: namespace: required value '''
E0128 18:44:28.217336   14608 file.go:45] Unable to read config path "random_non_existent_path": path does not exist, ignoring
--- FAIL: TestReadFromFileWithoutID (0.20 seconds)
	file_test.go:217: Expected update, timeout instead

--- a/pkg/kubelet/config/file_test.go
+++ b/pkg/kubelet/config/file_test.go
@@ -98,7 +98,7 @@ func TestUpdateOnNonExistentFile(t *testing.T) {
 			t.Fatalf("Expected %#v, Got %#v", expected, update)
 		}
 
-	case <-time.After(2 * time.Millisecond):
+	case <-time.After(time.Second):
 		t.Errorf("Expected update, timeout instead")
 	}
 }
@@ -165,7 +165,7 @@ func TestReadFromFile(t *testing.T) {
 			t.Fatalf("Expected %#v, Got %#v", expected, update)
 		}
 
-	case <-time.After(2 * time.Millisecond):
+	case <-time.After(time.Second):
 		t.Errorf("Expected update, timeout instead")
 	}
 }
@@ -213,7 +213,7 @@ func TestReadFromFileWithoutID(t *testing.T) {
 			t.Fatalf("Expected %#v, Got %#v", expected, update)
 		}
 
-	case <-time.After(2 * time.Millisecond):
+	case <-time.After(time.Second):
 		t.Errorf("Expected update, timeout instead")
 	}
 }
@@ -258,7 +258,7 @@ func TestReadV1Beta2FromFile(t *testing.T) {
 			t.Fatalf("Expected %#v, Got %#v", expected, update)
 		}
 
-	case <-time.After(2 * time.Millisecond):
+	case <-time.After(time.Second):
 		t.Errorf("Expected update, timeout instead")
 	}
 }
@@ -281,7 +281,7 @@ func TestReadFromFileWithDefaults(t *testing.T) {
 			t.Errorf("Unexpected UID: %s", update.Pods[0].ObjectMeta.UID)
 		}
 
-	case <-time.After(2 * time.Millisecond):
+	case <-time.After(time.Second):
 		t.Errorf("Expected update, timeout instead")
 	}
 }

